### PR TITLE
Add Socks5 proxy support

### PR DESCRIPTION
**Pull request by clienthax**
Usefull for ssh tunnels to cheap US vps's ;)

### CODE REVIEW

File: scripts/funidl.js [hunk 1]
```diff
@@ -14,6 +14,7 @@ const fs = require('fs');
 const shlp = require('sei-helper');
 const yargs = require('yargs');
 let request = require('request');
```
Comment (seiya-git) [inline scripts/funidl.js]: Please, change "let" to "const" here

File: scripts/funidl.js [hunk 2]
```diff
@@ -14,6 +14,7 @@ const fs = require('fs');
...
+var agent = require('socks5-https-client/lib/Agent');
```
Comment (seiya-git) [inline scripts/funidl.js]: do not use "var" here, "const" is better
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/const

File: scripts/funidl.js [hunk 3]
```diff
@@ -78,6 +79,8 @@ let argv = yargs
 	
 	// proxy
 	.describe('proxy','Set ipv4 http(s) proxy')
+	.describe('sockshost','Set socks5 host')
+	.describe('socksport','Set socks5 port')
```
Comment (seiya-git) [inline scripts/funidl.js]: Why not use single parameter for ip and port like with http(s) proxy?
And add checks to line 98.

File: scripts/funidl.js [hunk 4]
```diff
@@ -424,6 +427,16 @@ function getData(url,qs,proxy,useToken,auth){
 		options.proxy = 'http://'+argv.proxy;
 		options.timeout = 10000;
 	}
+
+	if(proxy && argv.sockshost && argv.socksport){
```
Comment (seiya-git) [inline scripts/funidl.js]: See previous comment.

File: scripts/funidl.js (hunk 3 above)
Comment (clienthax) [inline scripts/funidl.js]: I Thought that having to split it out with a string split would be a bit messier than just making two variables for it, + my knowledge of javascript is pretty limited, how would you suggest going about this?

Comment (seiya-git) [inline scripts/funidl.js]: Ok, I make changes by myself.

**Comment by seiya-git** [comment]
https://github.com/seiya-dev/funimation-downloader-nx/releases/tag/2.4.0
Thanks!

**Pull request closed without merging**